src/common/wallet-state.ts

```diff
@@ -368,11 +368,13 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
     setTotalBalance(some(balance))
   }
 
-  const getTimestamp = async (blockNumber: number | null): Promise<Date | null> => {
-    if (blockNumber === null) return null
-    const {timestamp} = await web3.eth.getBlock(blockNumber, true)
-    return _.isString(timestamp) ? fromUnixTime(parseInt(timestamp, 16)) : fromUnixTime(timestamp)
-  }
+  const getTimestamp = _.memoize(
+    async (blockNumber: number | null): Promise<Date | null> => {
+      if (_.isNil(blockNumber)) return null
+      const {timestamp} = await web3.eth.getBlock(blockNumber, true)
+      return _.isString(timestamp) ? fromUnixTime(parseInt(timestamp, 16)) : fromUnixTime(timestamp)
```

Comment:
If the `true` flag means "fetch full transactions" it should be omitted in this call, otherwise a lot of unnecessary data may be transferred and processed.